### ISSUE TITLE
1.1.1: error decoding compressed MessageSet: "unpack requires a bytes object of length 4" error

### ISSUE THREAD

**Issue by zoltan-fedor** [dpkp/kafka-python#710]
I have a compressed (snappy), partitioned topic, which is continuously being written and once an hour being read.

Unfortunately now reading got "stuck" with the following error:
`unpack requires a bytes object of length 4`

```
<FrameSummary file /data/apps/kabinet-dev/kabinet/jobs/avro.py, line 280 in run_job>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/consumer/group.py, line 844 in __next__>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/consumer/group.py, line 805 in _message_generator>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/consumer/fetcher.py, line 448 in __next__>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/consumer/fetcher.py, line 413 in _message_generator>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/consumer/fetcher.py, line 363 in _unpack_message_set>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/protocol/message.py, line 79 in decompress>,
<FrameSummary file /data/ap  ps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/protocol/message.py, line 148 in decode>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/protocol/message.py, line 50 in decode>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/protocol/message.py, line 50 in <listcomp>>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/protocol/types.py, line 79 in decode>,
<FrameSummary file /data/apps/kabinet-dev/kabinet/venv1/lib64/python3.5/site-packages/kafka/protocol/types.py, line 37 in decode>
```

This error gets thrown everytime when consumption reaches the same record.
Unfortunately when I manually reset the offset of that partition to "step over that record" then it finds other records elsewhere with the same issue, requiring manual offset reset again, otherwise it just stuck at that message.

**Comment by dpkp** [comment]
Is this the same as #713 ?

**Comment by zoltan-fedor** [comment]
Likely not, because this occurred in 1.1.1 and #713 had a different error.
I had to skip over a chunk of the feed in multiple partitions to get over these messages, so likely it was some kind of Kafka message or snappy compression problem.
Since then the feeds are running, but it still has a few 100M records to proces to catch up,, so if this issue comes back I will test it with 1.2.0 and the fix for #713

**Comment by dpkp** [comment]
ok, I have a hunch that this is fixed w/ 1.2.1. But will leave open until you have had a chance to verify

**Comment by zoltan-fedor** [comment]
Unfortunately (?) this issue hasn't come up since - after consuming a few 100M records and now upgraded to 1.2.1, so maybe won't come up anymore if that would really have the fix.

As my testing couldn't unearth another usecase to test against, closing this now.
If it comes up again, then will reopen.
